patches/api/0399-Add-extra-swing-method-for-convenience.patch

```diff
@@ -0,0 +1,29 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: SoSeDiK <mrsosedik@gmail.com>
+Date: Tue, 11 Oct 2022 22:35:56 +0300
+Subject: [PATCH] Add extra swing method for convenience
+
+
+diff --git a/src/main/java/org/bukkit/entity/LivingEntity.java b/src/main/java/org/bukkit/entity/LivingEntity.java
+index 4af5e8d0cba6555f7615e4e809d9aff221c0dc4d..3a5105ea55b8a007a61268ae354467b01562fc22 100644
+--- a/src/main/java/org/bukkit/entity/LivingEntity.java
++++ b/src/main/java/org/bukkit/entity/LivingEntity.java
+@@ -985,5 +985,18 @@ public interface LivingEntity extends Attributable, Damageable, ProjectileSource
+      * @param hurtDirection hurt direction
+      */
+     void setHurtDirection(float hurtDirection);
++
++    /**
++     * Makes this entity swing their hand.
++     *
++     * <p>This method does nothing if this entity does not
++     * have an animation for swinging their hand.
++     *
++     * @param hand hand to be swung, either {@link org.bukkit.inventory.EquipmentSlot#HAND} or {@link org.bukkit.inventory.EquipmentSlot#OFF_HAND}
++     */
++    default void swingHand(@NotNull org.bukkit.inventory.EquipmentSlot hand) {
++        if (hand == org.bukkit.inventory.EquipmentSlot.HAND) swingMainHand();
```

Comment:
add brackets to ifs and `this.` to instance method calls. also should have a preconditions check that one of the required values was passed

Reply:
Done & squished to a single commit.

Reply:
might have missed my edit on the other comment too, sorry :p

Reply:
Oh yea, just copypasted it without noticing 😅
Probably should be sleeping now instead of doing changes at 2 AM :)